src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageLoader.cs

```diff
@@ -20,21 +23,210 @@ public DefaultPageLoader(RazorCompiler compiler)
+
+            return results;
+        }
+
+        private static bool IsValidHandlerMethod(MethodInfo methodInfo)
+        {
+            // The SpecialName bit is set to flag members that are treated in a special way by some compilers
+            // (such as property accessors and operator overloading methods).
+            if (methodInfo.IsSpecialName)
+            {
+                return false;
+            }
+
+            // Overriden methods from Object class, e.g. Equals(Object), GetHashCode(), etc., are not valid.
+            if (methodInfo.GetBaseDefinition().DeclaringType == typeof(object))
+            {
+                return false;
+            }
+
+            if (methodInfo.IsStatic)
+            {
+                return false;
+            }
+
+            if (methodInfo.IsAbstract)
+            {
+                return false;
+            }
+
+            if (methodInfo.IsConstructor)
+            {
+                return false;
+            }
+
+            if (methodInfo.IsGenericMethod)
+            {
+                return false;
+            }
+
+            return methodInfo.IsPublic;
+        }
+
+        // Internal for testing
+        internal static HandlerParameterDescriptor[] CreateHandlerParameters(MethodInfo methodInfo)
+        {
+            var methodParameters = methodInfo.GetParameters();
+            var parameters = new HandlerParameterDescriptor[methodParameters.Length];
+
+            for (var i = 0; i < methodParameters.Length; i++)
+            {
+                var parameter = methodParameters[i];
+
+                parameters[i] = new HandlerParameterDescriptor()
+                {
+                    BindingInfo = BindingInfo.GetBindingInfo(parameter.GetCustomAttributes()),
+                    Name = parameter.Name,
+                    ParameterInfo = parameter,
+                    ParameterType = parameter.ParameterType,
+                };
+            }
+
+            return parameters;
+        }
+
+        // Internal for testing
+        internal static PageBoundPropertyDescriptor[] CreateBoundProperties(TypeInfo type)
+        {
+            var properties = PropertyHelper.GetVisibleProperties(type);
+
+            // If the type has a [BindPropertyAttribute] then we'll consider any and all public properties bindable.
+            var bindPropertyOnType = type.GetCustomAttribute<BindPropertyAttribute>();
+
+            var results = new PageBoundPropertyDescriptor[properties.Length];
+            for (var i = 0; i < properties.Length; i++)
+            {
+                var property = properties[i];
+                var bindingInfo = BindingInfo.GetBindingInfo(property.Property.GetCustomAttributes());
+
+                if (bindingInfo == null && bindPropertyOnType == null)
+                {
+                    continue;
+                }
+                
+                var bindPropertyOnProperty = property.Property.GetCustomAttribute<BindPropertyAttribute>();
```

Comment:
The `BindProperty` will be looked at by `BindingInfo` so we **will** get here if the property has one of those.